test/tests/functional/pbs_preemption.py

```diff
@@ -81,8 +84,99 @@ def test_preempted_never_run(self):
         self.server.expect(JOB, {ATTR_state: 'R'}, id=jidh)
 
         # The low priority job should be preempted
-        self.server.expect(JOB, {ATTR_state: 'S'}, id=jidl)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jidl)
+        self.scheduler.log_match(jidl + ";Job preempted by requeuing")
 
         # Check whether scheduler marked the preempted job as "will never run"
         self.scheduler.log_match(
             jidl + ";Job will never run", existence=False, max_attempts=5)
+
+    def test_preemption_checkpoint_never_run(self):
+        """
+        Test that a preempted job with checkpoint is not
+        marked as "Job will never run"
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname,
+                            expect=True)
+
+        # Create checkpoint
+        chk_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        self.chk_file = self.du.create_temp_file(body=chk_script)
+        self.du.chmod(path=self.chk_file, mode=0o755)
+        self.du.chown(path=self.chk_file, uid=0, gid=0, sudo=True)
+        c = {'$action': 'checkpoint_abort 30 !' + self.chk_file + ' %sid'}
+        self.mom.add_config(c)
+        self.attrs = {ATTR_l + '.select': '1:ncpus=1'}
+
+        # Set preempt order to checkpoint
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+
+        # Create high priority queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+
+        # submit a job to regular workq
```

Comment:
The priority of workq is lower than the new added expressq. New low priority queue is not added because that would be having an extra low priority queue in the setup. Also if there is some problem with the workq that would mean there is a problem with setup of PTL which this kind of test might catch.